COLLADABaseUtils/src/COLLADABUStringUtils.cpp

```diff
@@ -131,6 +204,65 @@ namespace COLLADABU
 
 
 
+	/*
+	The escape character is "_". Any XML name character that does not conform to the XML 1.0 spec (fourth edition) recommendation is escaped as _xHHHH_.
+	The HHHH string stands for the four-digit hexadecimal UCS-2 code for the character in most significant bit first order.
+	For example, the name Order Details is encoded as Order_x0020_Details.
+	*/
+	WideString StringUtils::ucs2Encode(const WideString  & sSrc)
+	{
+		const char DEC2HEX[16 + 1] = "0123456789ABCDEF";
+		wchar_t * pSrc = (wchar_t *)sSrc.c_str();
+		const int SRC_LEN = (const int)sSrc.length();
+		wchar_t * const pStart = new wchar_t[SRC_LEN * 7];
+		wchar_t * pEnd = pStart;
+		wchar_t * const SRC_END = pSrc + SRC_LEN;
+
+		for (; pSrc < SRC_END; ++pSrc)
+		{
+			*pEnd++ = '_';
+			*pEnd++ = 'x';
+			*pEnd++ = DEC2HEX[(*pSrc >> 12) & 0x0F];
+			*pEnd++ = DEC2HEX[(*pSrc >> 8) & 0x0F];
+			*pEnd++ = DEC2HEX[(*pSrc >> 4) & 0x0F];
+			*pEnd++ = DEC2HEX[*pSrc & 0x0F];
+			*pEnd++ = '_';
+		}
+
+		WideString sResult((wchar_t *)pStart, (wchar_t *)pEnd);
+
+		delete[] pStart;
+		return sResult;
+	}
+
+
+	String StringUtils::ucs2Encode(const String  & sSrc)
+	{
+		const char DEC2HEX[16 + 1] = "0123456789ABCDEF";
+		const unsigned char * pSrc = (const unsigned char *)sSrc.c_str();
+		const int SRC_LEN = (const int)sSrc.length();
+		unsigned char * const pStart = new unsigned char[SRC_LEN * 7];
+		unsigned char * pEnd = pStart;
+		const unsigned char * const SRC_END = pSrc + SRC_LEN;
+
+		for (; pSrc < SRC_END; ++pSrc)
+		{
+			*pEnd++ = '_';
```

Comment:
where did you find this ?
this does not look like XML escape sequence ?

&#DDD;  where DDD is the decimal number of the character
or 
&#xHH; where HH is the hexadecimal